crates/ide-db/src/defs.rs

```diff
@@ -932,3 +954,29 @@ impl TryFrom<DefWithBody> for Definition {
         }
     }
 }
+
+impl TryFrom<ItemContainer> for Definition {
+    type Error = ();
+    fn try_from(container: ItemContainer) -> Result<Self, Self::Error> {
+        match container {
+            ItemContainer::Trait(it) => Ok(it.into()),
+            ItemContainer::Impl(it) => Ok(it.into()),
+            ItemContainer::Module(it) => Ok(it.into()),
+            ItemContainer::ExternBlock() | ItemContainer::Crate(_) => Err(()),
+        }
+    }
+}
```

Comment:
Can we put this as a local closure inside `Definition::enclosing_definition`? The impl itself does not make too much sense to me (exposition wise)

Reply:
Done